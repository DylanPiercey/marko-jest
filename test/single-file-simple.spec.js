const path = require('path');
const { init, cleanup } = require('../');

const { render } = init(path.resolve(__dirname, './resources/single-file-simple/index.marko'));

describe('single-file component', () => {
  let renderResult;
  let component;

  afterEach(cleanup);

  describe('on rendering', () => {
    beforeEach(async () => {
      renderResult = await render({});
      ({ component } = renderResult);
    });

    it('should render the component as a button', () => {
      expect(component.el.matches('button')).toBeTruthy();
    });

    it('should render correctly', () => {
      expect(renderResult.getNodes()).toMatchSnapshot();
    });
  });

  describe('on triggering click handler', () => {
    beforeEach(async () => {
      renderResult = await render({});
      ({ component } = renderResult);

      component.el.click();
      component.update();
    });

    it('should update the element', () => {
      expect(renderResult.getNodes()).toMatchSnapshot();
    });

    it('should change the button label', () => {
      const buttonLabel = component.el.textContent;
      expect(buttonLabel).toEqual('DONE');
    });
  });
});
