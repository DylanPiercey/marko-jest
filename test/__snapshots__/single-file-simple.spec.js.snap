// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing single-file component on rendering should render correctly 1`] = `
Array [
  <button
    class="btn"
  >
    <span>
      Click me
    </span>
  </button>,
]
`;

exports[`testing single-file component on triggering click handler should update the element 1`] = `
Array [
  <button
    class="btn"
  >
    <span>
      DONE
    </span>
  </button>,
]
`;
